### PR TITLE
Fix 4x speed at wendigo

### PR DESCRIPTION
**Pull request by jlward**

### CODE REVIEW

File: battle/main.py [hunk 1]
```diff
@@ -2271,7 +2271,6 @@ def fullheal(target: int, direction: str):
 
 
 # Process written by CrimsonInferno
-@battle.utils.speedup_decorator
```
Comment (jlward) [inline battle/main.py:2274]: This fixes the wendigo fight, tested it on my machine.

File: battle/utils.py [hunk 2]
```diff
@@ -15,19 +15,20 @@ def _monitor_battle():
...
                 memory.main.set_game_speed(2)
                 fast = True
-        memory.main.set_game_speed(0)
+        memory.main.set_game_speed(old_game_speed)
 
     def wrapper(*args, **kwargs):
         if game_vars.get_battle_speedup():
             logger.debug(f"Speeding battle up: {func.__name__}")
-            monitor = threading.Thread(target=_monitor_battle)
+            monitor = threading.Thread(target=_monitor_battle, daemon=True)
```
Comment (jlward) [inline battle/utils.py:31]: This fixes the CTRL-C issue when you kill the TAS during combat when 4x speed is auto-enabled for combat. Also tested this locally.

File: nemesis/arenaPrep.py [hunk 3]
```diff
@@ -490,23 +490,23 @@ def advanced_battle_logic():
                             # Double Gemini, two different locations
                             if memory.main.get_use_items_slot(42) < 100:
                                 battle.main.use_item(
-                                    memory.main.get_use_items_slot(42), rikku_flee=False
+                                    memory.main.get_use_items_slot(42), rikku_flee=True
```
Comment (jlward) [inline nemesis/arenaPrep.py:493]: According to @coderwilson these all needed to be toggled, I have not actually tested this yet, but plan to either tonight or tomorrow.

File: memory/main.py [hunk 4]
```diff
@@ -2295,6 +2295,13 @@ def set_encounter_rate(set_val):
     process.write_bytes(key, set_val, 1)
 
 
+def get_game_speed():
+    global base_value
```
Comment (MrTyton) [inline memory/main.py:2299]: please use the helper funciton. details in #177

**Review by MrTyton — APPROVED** [review]

**Pull request merged**